mesonbuild/coredata.py

```diff
@@ -634,6 +634,12 @@ def set_option(self, key: OptionKey, value) -> None:
 
         if key.name == 'buildtype':
             self._set_others_from_buildtype(value)
+        elif key.name in ('wrap_mode', 'force_fallback_for'):
```

Comment:
Use a set for this `if z in {x, y}`

Reply:
OOC, does it make any difference in python interpreter (e.g. performance), or is it just semantically better?

In any case, changed now.

Reply:
If anything it should be a performance degradation because you need to compute hashes to the two strings to build the set and a third hash for the lookup (unless the interpreter has an optimization that skips that for small sets).

Reply:
In Python string comparison is O(n), and tuple/list membership tests are O(n), but a hash comparison is a single int comparison, and the lookup is log(n),

but, as they say, for proof:
```python
import timeit

# A pessimal list case
def a():
    return '71' in ['a', 'b', 'c', 'd', '71']

# Optimal list
def b():
    return 'a' in ['a', 'b', 'c', 'd', '71']
    
# set
def c():
    return '71' in {'a', 'b', 'c', 'd', '71'}

print('pessimal list: ', timeit.timeit(a))
print('optimal list: ', timeit.timeit(b))
print('set: ', timeit.timeit(c))
```
(timeit.timeit, for the unfamiliar, runs a function 10,000 times in a loop)

```
pessimal list:  0.08077858299657237
optimal list:  0.054498803001479246
set:  0.054652670994983055
```
So, the set lookup is *slightly* slower than the optimal case, but *much* faster than the pessimal case, and obviously that gets worse the more cases you add, at 10,000 members the list/tuple takes so long I gave up and canceld it, the set takes 0.0728 seconds. Unless your data is unhashable, you want a set for membership lookups, not a list or tuple. (There's no meaningful difference between a tuple and list for iteration speed)

Reply:
Thanks for the info, seems python sets are well implemented. In any case, it's sementically more right to use {} here, so I agree with that change.

My 2 cents on the off-topic:
For small sets, those numbers are pretty much meaningless. I don't know how python hash tables are implemented, but it can often be slower on small sets than a list/array (it famously is in GLib's GHashTable vs GPtrArray). After all, O(log(n)) < O(n) only for big (infinity) value of n.